### PR TITLE
MF-788: Add swr to allergies widget

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

This PR sets up the allergies widget to work with SWR. It introduces a reusable `useAllergies` hook that encapsulates fetching of allergies data from the backend using the SWR paradigm. 

Additionally, it introduces a call to the SWR [mutate](https://swr.vercel.app/docs/mutation) function which automatically revalidates the allergies widget data immediately after a new allergy is successfully recorded using the allergy form.

## Videos

https://user-images.githubusercontent.com/8509731/133847403-a3d341af-ac32-4350-99cf-76afd0ab97aa.mp4

https://user-images.githubusercontent.com/8509731/133847739-c997fbfb-61ee-4532-a479-c3848492aef1.mp4

https://user-images.githubusercontent.com/8509731/133848232-117352a3-41eb-4af8-8721-e70396357dfb.mp4

https://user-images.githubusercontent.com/8509731/133905597-6428136c-8dca-432a-8ed3-a799c005eb7c.mp4

## Related Issue

https://issues.openmrs.org/browse/MF-788

### CODE REVIEW

**Comment by HerbertYiga** [comment]
cc @FlorianRappl

**Comment by brandones** [comment]
Is this in progress (should be draft) or are you waiting on a review (request some reviewers)?

**Comment by denniskigen** [comment]
Ready for review @brandones

File: packages/esm-patient-allergies-app/src/constants.ts [hunk 1]
```diff
@@ -1 +1,3 @@
+export const ALLERGY_REACTION_CONCEPT = '162555AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA';
 export const patientAllergiesFormWorkspace = 'patient-allergy-form-workspace';
+export const allergiesToShowCount = 5;
```
Comment (brandones) [inline packages/esm-patient-allergies-app/src/constants.ts:3]: Matter of taste, but I'd just put this in the file it goes with. It's pretty specific to that component.

**Review by brandones — APPROVED** [review]
This rules. The code is shorter, has more robust state handling, and does the awesome SWR stuff. Thanks @denniskigen .

**Pull request merged**